lib/datadog/core/telemetry/schemas/base/v1/application.rb

```diff
@@ -0,0 +1,31 @@
+module Datadog
+  module Core
+    module Telemetry
+      module Schemas
+        module Base
+          module V1
+            # Describes attributes for application environment object
+            class Application
```

Comment:
Minor/more of a design thing: `Is::This::Class::Hierarchy::Getting::A::Bit::Too::Deep::And::Hard::To::Read`?

Such deep hierarchies are more common in Java, but they are seldom used in Ruby, because they can get really awkward to use (because of scoping rules, as well as not everyone using an IDE). 

For instance, even activesupport -- one of the most used libraries in Ruby and a core rails component ( https://github.com/rails/rails/tree/main/activesupport ) does not get 7 levels deep as we're doing here.

Could we perhaps shorten this to `Datadog::Core::Telemetry::Application`?

Reply:
That makes sense - would keeping the version still be useful (i.e. something like `Datadog::Core::Telemetry::V1::Application`)?

Reply:
> That makes sense - would keeping the version still be useful (i.e. something like Datadog::Core::Telemetry::V1::Application)?

That is an excellent question! We can't tell the future and how things will evolve, so if we think it's likely that a V2 and possibly others would come along, keeping `V1` would make a lot of sense; if, on the other hand, that's not very likely, then we'll live with the `V1` in there for years without ever needing it.

My past experience tells me we are probably not going to need to use a V2. There is a version in the API itself, because APIs are often a bit more strictly specified and less flexible, but in code it's common that we can support all api versions with the same class. (And even on the backend side, a common technique is also to implement multiple versions of the api with the same code as well).

So... my suggestion would be to not have the `V1`, but since this is 🔮 futurology, it may turn out to be wrong. And that's fine, I think -- the worst case is that we'll need to introduce a `Telemetry::V2::Application` at some point and deprecate `Telemetry::Application`.

See also YAGNI -> https://martinfowler.com/bliki/Yagni.html :)

Reply:
I agree with the idea of flattening the namespacing a bit here: `Schemas` and `Base` are a bit superfluous. Categorically interesting, but not necessary. These can be removed.

I think we should keep `V1`: the spec for this already describes the telemetry schema as "V1", and there are plans to iterate. It especially makes sense when the API is in the agent, and we make need to upgrade/downgrade protocols. Though I can appreciate YAGNI, it's not applicable here, and we benefit from a little "futurology" when it prevents a breaking change.

Reply:
Let me leave maybe a worse recommendation: `Datadog::Core::TelemetryV1`?